sourced/ml/models/id_splitter.py

```diff
@@ -0,0 +1,122 @@
+import logging
+import os
+import string
+from typing import List, Sequence, Tuple
+
+import keras
+from keras import backend
+from keras.preprocessing.sequence import pad_sequences
+from modelforge import Model, register_model
+import numpy as np
+
+from sourced.ml.algorithms.id_splitter.nn_model import (f1score, precision,
+                                                        recall)
+from sourced.ml.models.license import DEFAULT_LICENSE
+
+
+@register_model
+class IdentifierSplitterNN(Model):
+    """
+    Bi-Directionnal LSTM Model. Splits identifiers without need for a conventional pattern.
+    """
+    NAME = "id_splitter_nn"
+    VENDOR = "source{d}"
+    DESCRIPTION = "Model that contains source code identifier splitter BiLSTM weights."
+    LICENSE = DEFAULT_LICENSE
+
+    MAXLEN = 40
+    PADDING = "post"
+
+    def construct(self, model: "keras.models.Model" = None,
+                  session: "tf.Session" = None, maxlen: int = MAXLEN,
+                  padding: str = PADDING) -> "IdentifierSplitterNN":
+        assert model is not None
+
+        if session is not None:
+            backend.tensorflow_backend.set_session(session)
+        os.environ["TF_CPP_MIN_LOG_LEVEL"] = "1"
+
+        self._maxlen = maxlen
+        self._padding = padding
+        self._model = model
+        return self
+
+    @property
+    def model(self) -> "keras.models.Model":
+        """
+        Returns the wrapped keras model.
+        """
+        return self._model
+
+    def clear_keras(self):
+        """
+        Clears the keras session used to run the model
+        """
+        keras.backend.clear_session()
+
+    def _generate_tree(self) -> dict:
+        return {
+            "config": self._model.get_config(),
+            "weights": self._model.get_weights()
+            }
+
+    def _load_tree(self, tree: dict):
+        model = keras.models.Model.from_config(tree["config"])
+        model.set_weights(tree["weights"])
+        self.construct(model=model)
+
+    def _prepare_single_identifier(self, identifier: str, maxlen: int=MAXLEN, padding: int=PADDING,
```

Comment:
```suggestion
    def _prepare_single_identifier(self, identifier: str, maxlen: int = MAXLEN, padding: int = PADDING,
```
And the same everywhere lower